scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,378 @@
+    let commitHash := last.takeWhile (!·.isWhitespace)
+    let PRdescr := (last.drop commitHash.length).trim
+    return (commitHash, .trace {cls := `Commit} m!"{PRdescr}" #[m!"{commitHash}"])
+  let getFilesAtHash (hash : String) := do
+    let files ← runCmd s!"git ls-tree -r --name-only {hash} Mathlib/"
+    let h : Std.HashSet String := .ofList <| files.splitOn "\n"
+    return h
+  let (currentHash, currentPRdescr) ← getHashAndMessage 1
+  let currentFiles ← getFilesAtHash currentHash
+  msgs := msgs.push m!"{currentFiles.size} files at the current commit {currentPRdescr}"
+  let (pastHash, pastPRdescr) ← getHashAndMessage n
+  let pastFiles ← getFilesAtHash pastHash
+  msgs := msgs.push m!"{pastFiles.size} files at the past commit {pastPRdescr}"
+  let onlyPastFiles := pastFiles.filter fun fil ↦ fil.endsWith ".lean" && !currentFiles.contains fil
+  let noFiles := onlyPastFiles.size
+  msgs := msgs.push
+    m!"{noFiles} Lean file{if noFiles == 1 then "" else "s"} in 'Mathlib' that no longer exist:"
+  msgs := msgs.push "" ++ onlyPastFiles.toArray.map (m!"  {·}") |>.push ""
+  if onlyPastFiles.isEmpty then
+    logWarningAt (nc.getD ⟨tk⟩)
+      m!"All Lean files in Mathlib that existed {n} commits ago are still present.  \
+      Increase {n} to search further back!"
+    return
+  let mut suggestions : Array Meta.Tactic.TryThis.Suggestion := #[]
+  let ref := .ofRange {tk.getRange?.get! with stop := tk.getPos?.get!}
+  let dict ← mkRenamesDict 100
+  for fname in onlyPastFiles do
+    let fnameStx := Syntax.mkStrLit fname
+    let stx ← if let some newName := dict[fname]? then
+                let newNameStx := Syntax.mkStrLit newName
+                `(command|#create_deprecated_module $fnameStx rename_to $newNameStx)
+              else
+                `(command|#create_deprecated_module $fnameStx)
+    suggestions := suggestions.push {
+            suggestion := (⟨stx.raw.updateTrailing "hello".toSubstring⟩ : TSyntax `command)
+            }
+  let suggestionsText :=
+    if suggestions.size == 1 then ("the suggestion", "")
+    else (s!"any of the {suggestions.size} suggestions", ", so you can click several of them")
+  liftTermElabM do Meta.liftMetaM do
+    Meta.Tactic.TryThis.addSuggestions (origSpan? := some ref) (header := s!"Try these:\n\n\
+          Clicking on {suggestionsText.1} below will *not* remove the \
+          `#find_delete_files` command{suggestionsText.2}.\n") tk suggestions
+  logInfoAt tk <| .joinSep msgs.toList "\n"
+
+/-!
+If you already know the name of the file that you want to deprecate, then uncomment the
+`#create_deprecated_module` line below to get started, writing the file path as a string.
+* omitting `"a comment here"` is equivalent to using `"Auto-generated deprecation"`
+  while using the empty string `""` eliminates the comment entirely;
+* uncomment `write` only when you are satisfied that the deprecations look correct!
+  The command will also suggested this as a `Try this`.
```

Comment:
The grammar is off here: you mean to say "This command will also be suggested as a `Try this` suggestion by the `#find_deleted_files` command", right?

Reply:
Close enough: I meant "The command (`#create_deprecated_module ...`) will also suggested this (a version of itself with the uncommented `write`) as a `Try this`."

I'll use your wording, though, replacing `#find_deleted_files` by `#create_deprecated_module`.